spec/save_inventory/acyclic_graph_of_inventory_collections_spec.rb

```diff
@@ -80,18 +83,24 @@
 
         it 'creates a graph of InventoryCollections' do
           # Fill the InventoryCollections with data
-          add_data_to_inventory_collection(@data[:vms], @vm_data_1, @vm_data_12, @vm_data_2, @vm_data_4)
-          add_data_to_inventory_collection(@data[:miq_templates], @image_data_1, @image_data_2, @image_data_3)
-          add_data_to_inventory_collection(@data[:key_pairs], @key_pair_data_1, @key_pair_data_12, @key_pair_data_2,
-                                           @key_pair_data_3)
-          add_data_to_inventory_collection(@data[:hardwares], @hardware_data_1, @hardware_data_2, @hardware_data_12)
-          add_data_to_inventory_collection(@data[:disks], @disk_data_1, @disk_data_12, @disk_data_13, @disk_data_2)
-          add_data_to_inventory_collection(@data[:networks], @public_network_data_1, @public_network_data_12,
-                                           @public_network_data_13, @public_network_data_14, @public_network_data_2)
-          add_data_to_inventory_collection(@data[:flavors], @flavor_data_1, @flavor_data_2, @flavor_data_3)
+          {
+            :vms           => [@vm_data_1, @vm_data_12, @vm_data_2, @vm_data_4],
+            :miq_templates => [@image_data_1, @image_data_2, @image_data_3],
+            :key_pairs     => [@key_pair_data_1, @key_pair_data_12, @key_pair_data_2, @key_pair_data_3],
+            :hardwares     => [@hardware_data_1, @hardware_data_2, @hardware_data_12],
+            :disks         => [@disk_data_1, @disk_data_12, @disk_data_13, @disk_data_2],
+            :networks      => [@public_network_data_1, @public_network_data_12,
+                               @public_network_data_13, @public_network_data_14, @public_network_data_2],
+            :flavors       => [@flavor_data_1, @flavor_data_2, @flavor_data_3]
+          }.each_pair do |inventory_collection_name, data_arr|
+            data_arr.each do |data|
+              @persister.send(inventory_collection_name).build(data)
+            end
+          end
+
```

Comment:
@Ladas what do you think about this format?

Reply:
awesome